### PR TITLE
Submit work

### PR DESCRIPTION
**Pull request by Joshuaf91**

### CODE REVIEW

**Comment by Joshuaf91** [comment]
to test make sure you have these two empty directories magic_school_bus/assignments and magic_school_bus/work

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx [hunk 1]
```diff
@@ -8,8 +8,9 @@ const mapDispatchToProps = dispatch => (
   }, dispatch)
 );
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state, ownprops) => ({
```
Comment (samanthabretous) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: are you using `ownprops`?

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx [hunk 2]
```diff
@@ -18,48 +19,56 @@ class ShowAllAssignments extends Component {
...
-    this.setState({ assignment: idx, showAssignment: true });
+    this.setState({ assignment: idx, assignmentId, showAssignment: true });
   }
 
   assignmentsList() {
     const { assignments } = this.props;
     if (this.assignmentsExist()) {
       return assignments.map((ele, idx) =>
-        <li key={ele.id} value={idx} onClick={this.showAssignment}>
+        <li key={ele.id} value={idx} onClick={this.showAssignment.bind(this, ele.id)}>
```
Comment (samanthabretous) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: you are using bind here and in the constructor. you can pass the ele.id with an anonymous function ex:`onClick={() => this.showAssignment(ele.id, idx)}`

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx [hunk 3]
```diff
@@ -18,48 +19,56 @@ class ShowAllAssignments extends Component {
...
+    this.submitWork = this.submitWork.bind(this);
   }
 
   assignmentsExist() {
     return this.props.assignments;
   }
 
-  showAssignment(e) {
+  showAssignment(assignmentId, e) {
     const idx = e.target.value;
```
Comment (samanthabretous) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: see below on how to pass the index

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx [hunk 4]
```diff
@@ -18,48 +19,56 @@ class ShowAllAssignments extends Component {
...
+        <li key={ele.id} value={idx} onClick={this.showAssignment.bind(this, ele.id)}>
           {ele.instructions.slice(0, 49)}
         </li>);
     } else {
       return <li>NO ASSIGNMENTS FOR THIS CLASS</li>;
     }
   }
 
+  download(file) {
+    window.open(`/api/download/getAssignment?file=${file}`);
```
Comment (samanthabretous) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: Stack Overflow for Life

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx (hunk 2 above)
Comment (Joshuaf91) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: cool never tried this before i will implement it

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx [hunk 5]
```diff
@@ -18,48 +19,56 @@ class ShowAllAssignments extends Component {
...
       const { instructions, file, exercises, due } = this.props.assignments[assignment];
+      const { position } = this.props;
       return (<div>
         Instructions: {instructions}
-        <br/>
-       {/* this will display a clickable link to download file that instructor uploaded
-       File: {file}
-               <br/>*/}
+        <br />
+        {file ? <button onClick={this.download.bind(this, file)}>Download file </button> : 'no file for this assignment'}
```
Comment (samanthabretous) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: you can put the comment back in.

File: client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx [hunk 6]
```diff
@@ -68,6 +77,18 @@ class ShowAllAssignments extends Component {
     }
   }
 
+  submitWork() {
+    const { assignmentId } = this.state;
+    let currentPath = this.props.router.getCurrentLocation().pathname;
```
Comment (samanthabretous) [inline client/web_view/containers/ShowAllAssignments/ShowAllAssignments.jsx]: messy

File: client/web_view/containers/SubmitWork/SubmitWork.jsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    this.submit = this.submit.bind(this);
+  }
+
+  handelChange(e) {
+    this.setState({ zipFile: e.target.files[0], cantSubmit: false });
+  }
+
+  submit(e) {
+    e.preventDefault();
+    // push data into FormData because files must be sent in FormData
```
Comment (samanthabretous) [inline client/web_view/containers/SubmitWork/SubmitWork.jsx]: this doesn't make any sense. maybe rephrase what is going on

File: client/web_view/containers/SubmitWork/SubmitWork.jsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    const { zipFile } = this.state;
+    const { assignmentId, userId } = this.props;
+    const data = new FormData();
+    data.append('zipFile', zipFile, zipFile.name);
+    data.append('assignmentId', assignmentId);
+    data.append('userId', userId);
+    this.props.AsyncPostWork(data);
```
Comment (samanthabretous) [inline client/web_view/containers/SubmitWork/SubmitWork.jsx]: destructure

File: nodemon.json [hunk 9]
```diff
@@ -1,3 +1,3 @@
 {
-  "ignore": ["./client", "./repo", "./test", "./assignments"]
+  "ignore": ["./client", "./repo", "./test", "./assignments", "./work"]
```
Comment (samanthabretous) [inline nodemon.json]: yeah. I noticed that was happen too.

File: server/seed/user_seed.js [hunk 10]
```diff
@@ -10,7 +10,7 @@ module.exports = [
     firstName: 'Keesha',
     lastName: 'Franklin',
     email: 'kfranklin@msb.com',
-    username: 'kfranklin',
+    username: 'joshua',
```
Comment (samanthabretous) [inline server/seed/user_seed.js]: she's not joshua...

File: server/seed/work_seed.js [hunk 11]
```diff
@@ -1,61 +1,49 @@
 module.exports = [{
   zipFile: 'pede_lobortis_ligula.js',
```
Comment (samanthabretous) [inline server/seed/work_seed.js]: why are the zip extensions not .zip? we just need to seed the file correctly for production

Comment (Joshuaf91) [inline server/seed/work_seed.js]: so the work/assignments must be submitted on the web page to work because if the file does not exist it will throw an error
